lib/router/src/lib.rs

```diff
@@ -64,22 +64,21 @@ where
     cache: Mutex<Cache<Rec::Target, Stk::Value>>,
 }
 
-enum State<F, E>
+enum State<Req, Svc, E>
 where
-    F: Future,
+    Svc: svc::Service<Req>,
 {
-    Inner(F),
-    RouteError(E),
-    NoCapacity(usize),
-    NotRecognized,
-    Invalid,
+    NotReady(Req, Svc),
```

Comment:
These state transitions can be extracted to a shared `Oneshot` utility in `tower-util`. (not saying this PR should do it).